### PR TITLE
Authentication Refactor

### PR DESCRIPTION
**Pull request by joshgarde**
Closes https://github.com/podaac/swodlr-ui/issues/17

- Refactors swodlr-api to an OAuth2 resource server instead of an OAuth2 client
- Removes unnecessary code which was utilized as part of the older authentication flow
- Removes CMR integration which relied on older OAuth2 client interfaces
  - This interface will be moved to `raster-create`

### CODE REVIEW

File: src/main/java/gov/nasa/podaac/swodlr/security/config/WebSecurityConfig.java [hunk 1]
```diff
@@ -1,73 +1,40 @@
...
   public SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity http) {
     http
         .cors().and()
         .csrf().disable()
-        .authorizeExchange(authorize -> {
-          authorize
-              .pathMatchers("/config").permitAll()
-              .anyExchange().authenticated();
+        .authorizeExchange((exchange) -> {
+          exchange.pathMatchers("/edl/**").permitAll();
```
Comment (frankinspace) [inline src/main/java/gov/nasa/podaac/swodlr/security/config/WebSecurityConfig.java:28]: This totally removes the /config endpoint correct?
Does this still allow for an unauthenticated healthcheck for the ALB?

**Review by frankinspace — COMMENTED** [review]
Does this need to be coordinated with a UI update as well? Just want to understand if this gets deployed to SIT before any UI changes, will auth be broken during that time?

File: src/main/java/gov/nasa/podaac/swodlr/security/config/WebSecurityConfig.java (hunk 1 above)
Comment (joshgarde) [reply src/main/java/gov/nasa/podaac/swodlr/security/config/WebSecurityConfig.java:28]: Yes, it does. I'm not sure what we'd want to replace that with for the ALB

**Comment by joshgarde** [comment]
Yes, the new auth would need to be deployed on the frontend at the same time as the backend.

File: src/main/java/gov/nasa/podaac/swodlr/security/config/WebSecurityConfig.java (hunk 1 above)
Comment (frankinspace) [reply src/main/java/gov/nasa/podaac/swodlr/security/config/WebSecurityConfig.java:28]: Created another ticket for ALB concern: https://github.com/podaac/swodlr/issues/90

**Review by frankinspace — APPROVED** [review]

**Pull request merged**